scripts/generate.mjs

```diff
@@ -0,0 +1,78 @@
+import * as fs from "node:fs/promises";
+import { Octokit } from "octokit";
+
+const gh = new Octokit({
+  auth: process.env.GITHUB_TOKEN || process.env.GH_TOKEN,
+});
+
+const searchResults = gh.paginate.iterator(gh.rest.search.repos, {
+  q: "titanium in:topics language:objc language:swift language:java language:kotlin language:javascript",
+  per_page: 100,
+});
+
+const repositories = [];
+for await (const { data } of searchResults) {
+  const mappedData = await Promise.all(data.map(mapData));
+  repositories.push(...mappedData.filter((repo) => repo));
```

Comment:
I really think we need a setTimeout() just like I had to add them here: https://github.com/tidev/titanium-builds-regen-action/blob/main/index.js#L263. A 500ms pause isn't the end of the world.

Reply:
We don't need to handle that ourselves because the `octokit` package has [@octokit/plugin-retry](https://www.npmjs.com/package/@octokit/plugin-retry) and [@octokit/plugin-throttling](https://www.npmjs.com/package/@octokit/plugin-throttling) enabled by default. 

From the latter's docs `But the @octokit/plugin-throttling slows down your requests according to the official guidelines, so you don't get blocked before your quota is exhausted.`

Reply:
Yes, I see that in the readme, but I don't believe it in practice. This code is essentially the same as what is in the `tidev/titanium-builds-regen-action`, however we kept hitting API rate limits until I added `setTimeout()`'s on Jan 6, 2023. Since then, no issues. https://github.com/tidev/downloads-www/actions?page=3

So, I'm cautiously ok with this, but I have a strong feeling we will be adding `setTimeout()`'s when both the module search and the downloads regen are borked due to API rate limiting.

Reply:
It's not the same though, `getOctokit` from `@actions/github` returns an instance of `@octokit/core` ([this](https://github.com/octokit/core.js)) whereas here I'm using `octokit` ([this](https://github.com/octokit/octokit.js)) which wraps `@octokit/core` and adds a bunch of other stuff (like throttling and retry plugins)

Reply:
Ooooooh. So why on Earth did I use `@actions/github`?

Reply:
tbf for an action that makes sense I guess as it's just plug and play and picks up auth automatically, where I think for plain octokit you have to handle it

Reply:
But isn't the "auth" part just initializing octokit with the token?

Reply:
Don't give away the secret sauce! TBH, I bet most of it is just "marketing" really